src/bake/DevServer.zig

```diff
@@ -1704,6 +1706,8 @@ const ResponseAndMethod = struct {
     method: bun.http.Method,
 };
 
+var next_inspector_build_id = std.atomic.Value(i32).init(0);
```

Comment:
what are build IDs needed for? isn't a better system just to notify with messages "there is a build happening" and then "*all* builds are complete". note that only one build runs at a time, so tagging each with an id doesn't really make sense.

if this is truly needed, it should go at the top so people reading this file are aware there is global state (this is the first piece of global state)

Reply:
You could imagine a stateful message "Bundle started" updating to "Bundling..." and later updating to "Bundle completed/failed"

Reply:
build IDs aren't needed or help with this since there is only ever one build at once. additionally, multiple successful builds in a row probably should appear as one build to the inspector (the case where a second build is already queued, due to a save-during-build).

Reply:
> only ever one build at once

While this is technically true currently, messages sent between processes and the network make this not true in practice for this usecase.

Reply:
> messages sent between processes and the network make this not true in practice for this usecase.

TCP/WebSocket/IPC guarantee ordered messages

since the start message is in startAsyncBundle, another "build started" event. similar handling is done with the testing hook `testing_batch_events`, and in the `batchChanges` function in javascript. this doesn't use build ids, because once a build starts the only next message is "build success" or "build fail". though, if multiple dev servers exist, this would get confused. there probably should be a "dev server id" so that the inspector can also understand which HmrSocket tie to which DevServer.

these build ids just give you more room to mess up.